staging/src/k8s.io/client-go/tools/cache/controller.go

```diff
@@ -270,10 +270,8 @@ func DeletionHandlingMetaNamespaceKeyFunc(obj interface{}) (string, error) {
 //  * lw is list and watch functions for the source of the resource you want to
 //    be informed of.
 //  * objType is an object of the type that you expect to receive.
-//  * resyncPeriod: if non-zero, will re-list this often (you will get OnUpdate
-//    calls, even if nothing changed). Otherwise, re-list will be delayed as
-//    long as possible (until the upstream source closes the watch or times out,
```

Comment:
removed, as clean shutdown of Watch doesn't seem to be causing re-listing

Reply:
Leave this comment. a watch connection closing does cause a relist when it is re-established if the last watched revision is no longer available

Reply:
I see, but previous comment was also misleading. These re-lists have nothing to do with resyncPeriod, right? Maybe it is better to add dedicated section about possible causes of List events to reflector?

List of re-list causes would be:
 1.returned non-nil Error from `Store.Resync`
 2. most of errors from Watch call
 
what do you think?

Reply:
oh i see , we are trying to differentiate between resyncs(guided by resync period) and re-lists guided by broken watches ? Is it necessary to differentiate them  ?

Reply:
> Is it necessary to differentiate them ?

I think so. There is a popular belief (fuelled by imprecise comments this PR tries to fix) that resync implies relists, for instance prometheus-operator https://github.com/coreos/prometheus-operator/blob/a174c8b1281219526db395a669d65a5bac7fe425/pkg/prometheus/operator.go#L269-L278

Reply:
it is useful to include the information that re-list also triggers calling OnUpdate, even if nothing changed.

suggest:
> if non-zero, will resync this often (you will get OnUpdate
> calls, even if nothing changed). if zero, OnUpdate will only be called
> when a resource changes, or when a re-list occurs (when the upstream
> source closes the watch or times out).